### PR TITLE
[Avatar] Expose `onError` hook

### PR DESCRIPTION
**Pull request by whizkydee**
### WHY are these changes introduced?

Due to surface area constraints, in our app, we use custom a user menu — the activator is a standalone avatar within a button which toggles a `Popover`. The popover includes an avatar and the full name of the user in question, as opposed to displaying the full name alongside the avatar directly on the top bar.

For Identity users that haven't uploaded an avatar, we attempt to grab an image from gravatar based on their `uuid` which eventually resolves to their email — that request can either return the image or `404` if the account doesn't exist.

The problem with that is each time the user menu activator is clicked, the popover is re-rendered which results in extraneous requests to the same url that previously failed in an attempt to retrieve the gravatar.

### WHAT is this pull request doing?

This adds an `onError` hook to the `Avatar` component, which provides consumers with an avenue to make decisions when the HTTP request for an avatar fails. In our case, we'll simply prevent subsequent requests to the same url.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page, Avatar} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <Avatar source="invalidUrl" onError={() => console.log('bummer')} />
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Avatar/Avatar.tsx [hunk 1]
```diff
@@ -31,13 +31,16 @@ export interface AvatarProps {
   customer?: boolean;
```
Comment (whizkydee) [inline src/components/Avatar/Avatar.tsx:31]: Thought — Should this be the default behaviour of the `Avatar` component?

**Comment by alex-page** [comment]
@whizkydee I am wondering if the error could be handled in your app before using the component? It feels like the wrong level of abstraction. Interested to know what blockers exist there?

**Comment by whizkydee** [comment]
@alex-page The problem is we have know way of knowing that the image errored, so we can't handle it. This hook is meant to inform us.

**Review by alex-page — APPROVED** [review]
Happy to ship this, but I still think this check should happen before rendering the `Avatar` component.

**Review by kyledurand — APPROVED** [review]
I think this makes sense to add but also agree with Alex that the best UX would be to determine if the image request 404'd before trying to render the Avatar. 

Sounds like your situation is a bit different with the re-rendering on click though so I think if this works for your use case then 🚢

**Pull request merged**